### PR TITLE
fix(hc): Adds fix for evaluation context builder error handler, adds feature option

### PR DESCRIPTION
**Pull request by GabeVillalobos**
## Feature Option
Adds a new `feature` option which will be used to house internal feature flag data coming from options-automator.

## Evaluation Context Fix
Updates the context builder's `add_error_handler` method to return a reference to self in order for more builder methods to be chained.

### CODE REVIEW

File: src/sentry/options/defaults.py [hunk 1]
```diff
@@ -1793,6 +1793,10 @@
     "hybrid_cloud.disable_relative_upload_urls", default=False, flags=FLAG_AUTOMATOR_MODIFIABLE
 )
 
+# Flagpole Rollout
+register("features", default={}, flags=FLAG_AUTOMATOR_MODIFIABLE)
```
Comment (markstory) [inline src/sentry/options/defaults.py:1797]: Would we have all feature flags stored as keys in this dictionary? Another approach could be to have one option per feature flag, and prefix options with `features.` That would let us load feature data as required and not evict the entire feature cache when any feature changes.

File: src/sentry/options/defaults.py [hunk 2]
```diff
@@ -1793,6 +1793,10 @@
...
+register("hybrid-cloud.flagpole_rollout_phase", default=0, flags=FLAG_AUTOMATOR_MODIFIABLE)
```
Comment (markstory) [inline src/sentry/options/defaults.py:1798]:
```suggestion
register("hybridcloud.flagpole_rollout_phase", default=0, flags=FLAG_AUTOMATOR_MODIFIABLE)
```

Could we normalize on `hybridcloud` or `hybrid_cloud`?

File: src/sentry/options/defaults.py (hunk 1 above)
Comment (GabeVillalobos) [reply src/sentry/options/defaults.py:1797]: Hmm wouldn't this also require individually registering each feature option with a default as well? Either that or we fully relax the concept of pushing unknown options to sentry (or regex match against feature_*?)

File: src/sentry/options/defaults.py (hunk 2 above)
Comment (markstory) [reply src/sentry/options/defaults.py:1798]: I'm not too fussy on which one we use as we have two already :cry:

File: src/sentry/options/defaults.py (hunk 1 above)
Comment (GabeVillalobos) [reply src/sentry/options/defaults.py:1795]: Revisiting this finally, I think a single features dictionary is fine to start. We'll work on splitting this into individual features as needed if we encounter perf issues retrieving or setting this monolithic features option.

File: src/sentry/options/defaults.py (hunk 2 above)
Comment (GabeVillalobos) [reply src/sentry/options/defaults.py]: Should be fixed now

**Review by markstory — APPROVED** [review]

**Pull request merged**